### ISSUE TITLE
Follower leveling enhancements

### ISSUE THREAD

**Issue by Oggen** [TLDRMissions/TLDRMissions#390]
**TLDR Version of your idea**
TLDR version: Send bonus xp missions with less than min followers if that's all that remains, and treat non bonus xp missions above a certain xp threshold as bonus xp missions.

**Slightly longer version**
Part 1:
In order to send bonus xp missions with more than a single follower, you can set special treatment and choose a minimum follower count for bonus xp missions. But if you've run low on remaining followers that aren't max level, it won't send any remaining bonus xp missions. For example, if I have 8 followers that need to be leveled and there are 2 bonus xp missions, I could set minimum bonus xp follower count to 5 and it would only send 1 bonus xp mission, leaving the last 3 followers out when it would make sense to send them on the 2nd one.

Part 2:
Some missions, especially crafting material missions, have pretty good xp where I would like to send followers that need to level up on them after I've run out of bonus xp missions. It would be great to have a slider where I could say "treat non bonus xp missions that reward more than 1000 xp as bonus xp missions" so they could follow the special treatment (putting them at a corresponding mission priority to the total xp reward amount).

These two enhancements would make leveling followers even easier!

**Comment by teelolws** [comment]
For Part 1: if you set the "minimum count" to 1, it will still send 5 on your first mission, and then 3 on the second mission.

For Part 2: the "then send remaining followers for XP" option already does this?

**Comment by Oggen** [comment]
Part 1: Well I could've SWORN that I saw something different, but it looks like it does indeed work how I was hoping. Excellent, and my apologies for missing this.

Part 2: The option you mention partially accomplishes this, but it's the lowest priority which can mess things up. If my goal is to level up my lowest followers then I will put bonus follower xp missions as top priority and set lowest level followers first. But if there aren't any follower xp missions then it'll send my lowest level followers on the next highest priority missions which might reward very low xp. By the time it has exhausted all the mission types I care about then it's possible there won't be any more low-level followers available to run, say, a leftover 1500 xp cooking material mission which has an item reward I don't care about but a juicy xp reward I would consider high priority. What I'm proposing is a slider to effectively say "treat any mission with an xp reward larger than this amount as a bonus follower xp mission in terms of priority and special treatment." One extra consideration is that currently the option you're describing intends to fail the mission, but sometimes the missions with good xp have items that are useful and would be preferred to not fail, so perhaps if a non bonus xp mission is elevated to bonus xp mission priority then it would decide whether to succeed or fail based on whether it falls into another checked priority? Sorry for the wall of text, I hope I described this well enough and that this makes any sense. The short of it is: I'd love to be able to prioritize good xp missions, whether they're categorized as "bonus xp" or not.

Minor bonus part 3: In the special treatment of follower xp missions you can set to choose only followers that are required to level troops. My understanding is this will ignore extra xp missions if there are no available followers that affect troop level, but it would be great to have this option instead (or in addition) just prioritize those followers and then fall back to highest or lowest level followers afterwards.

Thank you so much for reading this! Your project is truly amazing, such a game-changer for something that otherwise becomes so repetitive and chore-like.

**Comment by teelolws** [comment]
For Part 3: if you just set it to "highest" then it will accomplish what you want. The followers that affect troop level *are* the highest level followers.

I'll consider your other request.

**Comment by Oggen** [comment]
This is definitely getting too picky here but it seems common for it to not be the highest that affects the troop level. I think troop level is the median follower level, so if you pick the highest troop level setting then it could keep leveling up the top 40% of followers which would never raise the median and the troop level would stay the same. Thank you for considering!

**Comment by teelolws** [comment]
Heres what I'm settling on: 

![image](https://user-images.githubusercontent.com/49792789/176847859-80d0c8ba-4ef1-4978-8a80-4527928f842e.png)

It will respect your "Use Special Treatment for Follower XP Missions" settings, treating it as a bonus XP mission.
